docs/tutorials/web/ep-webgpu.md

```diff
@@ -0,0 +1,195 @@
+const bufferSize = (10 * 1000) /* number of elements */ * 4 /* bytes per element */;
+const device = ort.env.webgpu.device;
+const myPreAllocatedBuffer = device.createBuffer({
+    usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST | GPUBufferUsage.STORAGE,
+    size: Math.ceil(size / 16) * 16 /* align to 16 bytes */
+});
+
+const myPreAllocatedOutputTensor = Tensor.fromGpuBuffer(myPreAllocatedBuffer, {
+  dataType: 'float32',
+  dims: [10, 1000]
+});
+
+// ...
+
+// Run the session with fetches
+const feeds = { 'input_0': myInputTensor };
+const fetches = { 'output_0': myPreAllocatedOutputTensor };
+const results = await mySession.run(feeds, fetches);
+
+```
+
+By specifying the output tensor in the fetches, ONNX Runtime Web will use the pre-allocated buffer as the output buffer. If there is a shape mismatch, the `run()` call will fail.
+
+### Specify the output data location
+
+If you don't want to use pre-allocated GPU tensors for outputs, you can also specify the output data location in the session options:
+
+```js
+const mySessionOptions1 = {
+  ...,
+  // keep all output data on GPU
+  preferredOutputLocation: 'gpu-buffer'
+};
+
+const mySessionOptions2 = {
+  ...,
+  // alternatively, you can specify the output location for each output tensor
+  preferredOutputLocation: {
+    'output_0': 'cpu',         // keep output_0 on CPU. This is the default behavior.
+    'output_1': 'gpu-buffer'   // keep output_1 on GPU buffer
+  }
+};
+```
+
+By specifying the config `preferredOutputLocation`, ONNX Runtime Web will keep the output data on the specified device.
+
+See [API reference: preferredOutputLocation](https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html#preferredOutputLocation) for more details.
+
+## Notes
+
+### Zero-sized tensors
+
+If a tensor's shape contains 1 or more dimensions with size 0, the tensor is considered as a zero-sized tensor. Zero-sized tensors do not have any data, so the data location is not applied. ONNX Runtime Web always treats zero-sized tensors as CPU tensors. To create a zero-sized tensor, you can use the following code:
+
+```js
+const zeroSizedTensor = new ort.Tensor('float32', [], [3, 256, 0, 64]);
+```
+
+### GPU tensor life cycle management
+
+It is important to understand how the underlying GPU buffer is managed so that you can avoid memory leaks and improve buffer usage efficiency.
+
+A GPU tensor is created either by user code or by ONNX Runtime Web as model's output.
+- When it is created by user code, it is always created with an existing GPU buffer using `Tensor.fromGpuBuffer()`. In this case, the tensor does not "own" the GPU buffer. It is user's responsibility to make sure the underlying buffer is valid during the inference, and call `buffer.destroy()` to dispose the buffer when it is no longer needed. Using a GPU tensor with a destroyed GPU buffer will cause the session run to fail.
```

Comment:
Do we need to call tensor.dispose() for this situation?

Reply:
No. do we want to mention this explicitly here?

Reply:
Better, as it's a difference between two similar usages.